docs/source/ts-ssdk/validation.rst

```diff
@@ -0,0 +1,137 @@
+.. _TS SSDK validation:
+
+=======================================
+Smithy TypeScript Server SDK Validation
+=======================================
+
+Smithy defines a set of :ref:`constraint traits <constraint-traits>` that can be used to restrict the set of allowed
+values for a shape. These traits do not change the type generated; the type will always correspond with the Smithy type.
+This allows Smithy models to evolve over time, so that model changes will not cause changes that break type checks.
+
+Smithy Server SDKs will only validate input values against constraint traits. While output structures can have
+constraint traits applied, they will never be evaluated by the Server SDK directly. This is because the failure would
+be too late in the process. At the point of validation, the Server SDK's could only to return an
+:ref:`unexpected failure <TS SSDK internal-failure-exception>` to the user, and changes made by the operation
+implementation would neither be returned to the user nor rolled back.
+
+Default Validation
+==================
+
+The Server SDK validates inputs by default if every operation in the service has the
+`smithy.framework#ValidationException`_ error associated. This can be accomplished by associating the error directly
+with the :ref:`service <service>`, as well. In this mode, the constraint traits built into Smithy are applied to each
+input and an error is generated and automatically returned if any are violated. ``ValidationException`` contains an
+entry for each constraint violation encountered in the input, along with a pointer to the value that failed validation,
```

Comment:
Should "a pointer to the value that" be "a pointer to the member thats value failed"?